### ISSUE TITLE
add contents to doc/declare_agents.md

### ISSUE THREAD

**Issue by maeste** [wise-agents/wise-agents#150]

**Comment by maeste** [comment]
@kabir I've reassigned this to you because it's basically  the documentation for how to write yaml file and it's related to the work you are doing on validating them

**Comment by kabir** [comment]
Makes sense. I thought I should write something about the env var substitution but it is probably better to wait until it is all working and validated.
